integration-libs/punchout/core/facade/punchout.service.ts

```diff
@@ -42,51 +45,62 @@ export class PunchoutService implements PunchoutFacade {
    * Redirect to Punchout Error page if error occurs
    */
   protected getPunchoutSessionCommand: Command<
-    { sessionId: string },
+    PunchoutSessionInput,
     PunchoutSession
   > = this.commandService.create((payload) => {
-    if (!payload?.sessionId) {
+    if (!payload?.punchoutSessionId) {
       this.displayErrorPage();
       return throwError(() => new Error('Punchout Session Id missing'));
     }
-    return this.punchoutConnector.getPunchoutSession(payload.sessionId).pipe(
-      map((punchoutSession) => {
-        if (
-          !punchoutSession?.token?.accessToken ||
-          !punchoutSession?.customerId
-        ) {
-          throw new Error('Punchout login info missing');
-        }
-        return punchoutSession;
-      }),
-      switchMap((punchoutSession) => {
-        return forkJoin({
-          punchoutSession: of(punchoutSession),
-          logout: this.punchoutAuthService.logout(),
-        });
-      }),
-      map(({ punchoutSession }) => {
-        if (punchoutSession?.token?.accessToken) {
-          this.punchoutAuthService.loginWithToken(
-            punchoutSession.token.accessToken,
-            punchoutSession.customerId
-          );
-          this.routeToTargetPage(punchoutSession);
-        } else {
-          throw new Error('Punchout Access Token missing');
-        }
+    return this.punchoutConnector
+      .getPunchoutSession(payload.punchoutSessionId)
+      .pipe(
+        map((punchoutSession) => {
+          if (
+            !punchoutSession?.token?.accessToken ||
+            !punchoutSession?.customerId
+          ) {
+            throw new Error('Punchout login info missing');
+          }
+          return punchoutSession;
+        }),
+        switchMap((punchoutSession) => {
+          return forkJoin({
+            punchoutSession: of(punchoutSession),
+            logout: this.punchoutAuthService.logout(),
+          });
+        }),
+        map(({ punchoutSession }) => {
+          if (punchoutSession?.token?.accessToken) {
+            this.punchoutAuthService.loginWithToken(
+              punchoutSession.token.accessToken,
+              punchoutSession.customerId
+            );
 
-        return punchoutSession;
-      }),
-      catchError((error) => {
-        this.displayErrorPage();
-        return throwError(() => new Error(error));
-      })
-    );
+            this.punchoutStoreService.setPunchoutState({
+              punchoutSessionId: payload.punchoutSessionId,
+              punchoutSession: { ...punchoutSession },
+            });
+            if (!payload?.isPageRefresh) {
```

Comment:
New optional flag 'isPageRefresh'  prevents to navigate elsewhere.
Essentially used in situation such' refreshing the page' or closing the tab and back.